code/game/machinery/bank_machine.dm

```diff
@@ -32,58 +42,46 @@
 		var/obj/item/holochip/H = I
 		value = H.credits
 	if(value)
-		var/datum/bank_account/D = SSeconomy.get_budget_account(ACCOUNT_CAR_ID)
-		if(D)
-			D.adjust_money(value)
-			to_chat(user, span_notice("You deposit [I]. The Cargo Budget is now $[D.account_balance]."))
+		if(HAS_TRAIT(SSstation, STATION_TRAIT_UNITED_BUDGET))
+			var/datum/bank_account/united_budget = SSeconomy.get_budget_account(ACCOUNT_CAR_ID)
+			united_budget.adjust_money(value)
+			to_chat(user, span_notice("You deposit [value] into a station budget account.</span>"))
+		else
+			var/money_amount_modulo = (value % length(list_of_budgets))
+			var/rounded_money_amount = ((value - money_amount_modulo) / length(list_of_budgets))
+			if(money_amount_modulo)
+				var/datum/bank_account/first_budget_card = list_of_budgets[1]
+				first_budget_card.adjust_money(money_amount_modulo) //If we have an indivisible amount of money, dump it in the first budget.
+			for(var/datum/bank_account/budget_department_id as anything in list_of_budgets)
+				budget_department_id.adjust_money(rounded_money_amount)
+			to_chat(user, span_notice("You deposit [value] into all station budgets."))
```

Comment:
I do have a question about if we actually want this behaviour? Currently, when the cargo budget is lost/stolen, this machine is the only method of providing money into the cargo's bank account. I would like either:
- All money to go into the cargo bank account.
- The option to choose where the money is going to go to, or all budgets.